doit/dependency.py

```diff
@@ -36,6 +36,9 @@ def get_file_md5(path):
     @param path: (string) file path
     @return: (string) md5
     """
+    if os.path.isdir(path):
```

Comment:
actually this is probably not a good location for doing this check. `check_modified` was carefully implemented to avoid getting `file_stat` more than once. since this is a relatively expensive operation.